ttnn/cpp/ttnn/operations/ccl/all_gather/all_gather_pybind.cpp

```diff
@@ -75,6 +99,37 @@ void py_bind_all_gather(pybind11::module& module) {
             >>> output = ttnn.all_gather(tensor, dim=0)
 
         )doc");
+
+    /*
+    detail::bind_line_all_gather(
+        module,
+        ttnn::line_all_gather,
+        R"doc(line_all_gather(input_tensor: ttnn.Tensor, dim: int, *, num_links: int = 1, memory_config: Optional[ttnn.MemoryConfig] = None) -> ttnn.Tensor
+
+        Performs an all-gather operation on multi-device :attr:`input_tensor` across all devices.
+        Args:
+            * :attr:`input_tensor` (ttnn.Tensor):
+                multi-device tensor
+            * :attr:`dim` (int):
+                Dimension to perform the all-gather operation on.
+                After the all-gather operation, the size of the :attr:`dim`
+                dimension will larger by number of devices in the line.
+            * :attr:`cluster_axis` (int):
+                Provided a MeshTensor, the axis corresponding to MeshDevice
+                to perform the line-all-gather operation on.
+            * :attr:`mesh_device` (MeshDevice):
+                Device mesh to perform the line-all-gather operation on.
```

Comment:
@SeanNijjar what can we do with this? can we have two bindings for ring and line or add this commented code as common binding by mentioning the args like cluster axis are only for line?